core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/protocol/restxml/XmlPayloadMarshaller.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.protocol.restxml;
+
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_BIG_DECIMAL;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_BOOLEAN;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_DOUBLE;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_FLOAT;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_INSTANT;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_INTEGER;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_LONG;
+import static software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter.FROM_STRING;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.internal.protocol.json.ValueToStringConverter;
+import software.amazon.awssdk.core.protocol.MarshallLocation;
+import software.amazon.awssdk.core.protocol.SdkField;
+import software.amazon.awssdk.core.protocol.SdkPojo;
+import software.amazon.awssdk.core.protocol.traits.ListTrait;
+import software.amazon.awssdk.core.protocol.traits.LocationTrait;
+import software.amazon.awssdk.core.protocol.traits.MapTrait;
+import software.amazon.awssdk.core.util.SdkAutoConstructList;
+import software.amazon.awssdk.core.util.SdkAutoConstructMap;
+
+@SdkInternalApi
+public class XmlPayloadMarshaller {
+
+    public static final XmlMarshaller<String> STRING = new BasePayloadMarshaller<String>(FROM_STRING) {
+        @Override
+        public void marshall(String val, XmlMarshallerContext context, String paramName,
+                                SdkField<String> sdkField, ValueToStringConverter.ValueToString<String> converter) {
+            context.xmlGenerator().xmlWriter().startElement(paramName).value(val).endElement();
```

Comment:
In JSON we emit the field name in the base class. Could we do something similiar here? I.e. start and end in the base and emit just the value in the impl.

Reply:
For list, we should add the top level startElement/endElement only if it is not flattened. We don't have that information is base class, hence we can't move the start/end tags to base.

Reply:
Could we just have a separate impl for list and maps? Seems like we are duplicating a lot of effort just to handle two of the types.